### PR TITLE
resolve emacs click positioning issue w/mathjax

### PR DESCRIPTION
**Pull request by kevinushey**
This PR should resolve an issue where attempts to click on a line in the Ace editor would occasionally select the character immediately above. Two things are required to reproduce:

1. Emacs keybindings must be active,
2. A Mathjax line widget must be showing above the click position.

Gif for demonstration:

mathjax

The fix here just removes the Emacs-specific click offset handling; in theory this implies that clicks might behave a little less Emacs-y but I think it's unlikely most users will notice.

For reference, the default `screenToTextCoordinates` behavior is defined here: https://github.com/ajaxorg/ace/blob/f757c8568fba3e59392383e332519e5570f54e1e/lib/ace/virtual_renderer.js#L1415-L1425

The chief difference I can see is the use of `Math.round()` versus `Math.floor()`, so my guess is that something about how we are fudging the offsets for line widgets is causing issues -- however, this seems like the simpler fix.

### CODE REVIEW

**Comment by kevinushey** [comment]
Thinking a bit more -- it might be better just to bring the y-position calculation in line with the default (I notice the x-position of the click seems 'off' with this patch)

**Comment by kevinushey** [comment]
Rebased now -- we no longer fudge the vertical positioning when in Emacs mode, but leave the horizontal positioning as-is.

File: src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/ace/keybinding-emacs-uncompressed.js [hunk 1]
```diff
@@ -638,9 +638,8 @@ var screenToTextBlockCoordinates = function(x, y) {
     var col = Math.floor(
         (x + this.scrollLeft - canvasPos.left - this.$padding) / this.characterWidth
     );
-    var row = Math.floor(
-        (y + this.scrollTop - canvasPos.top) / this.lineHeight
-    );
+
+    var row = (y + this.scrollTop - canvasPos.top) / this.lineHeight;
 
     return this.session.screenToDocumentPosition(row, col);
```
Comment (jmcphers) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/ace/keybinding-emacs-uncompressed.js]: Is it OK to return a fractional row here, or should we `Math.round` above?

Comment (kevinushey) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/ace/keybinding-emacs-uncompressed.js]: It should be okay. Note that this logic mirrors what Ace does in the default `screenToTextCoordinats()` function here: https://github.com/ajaxorg/ace/blob/f757c8568fba3e59392383e332519e5570f54e1e/lib/ace/virtual_renderer.js#L1422-L1424

Comment (jmcphers) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/ace/keybinding-emacs-uncompressed.js]: Ah, great. :-)

**Pull request merged**